ow_faults/include/ow_faults/FaultDetector.h

```diff
@@ -43,9 +43,139 @@ class FaultDetector
   FaultDetector& operator= (const FaultDetector&) = delete;
 
 
+  enum class ComponentFaults : uint {
+    Hardware = 1, 
+    JointLimit = 2,
+    TrajectoryGeneration = 2,
+    Collision = 3, 
+    Estop = 4, 
+    PositionLimit = 5, 
+    TorqueLimit = 6, 
+    VelocityLimit = 7, 
+    NoForceData = 8
+    };
+
+  //system
+  static constexpr std::bitset<10> isSystem{                0b00'0000'0001 };
+  static constexpr std::bitset<10> isArmGoalError{          0b00'0000'0010 };
+  static constexpr std::bitset<10> isArmExecutionError{     0b00'0000'0100 };
+  static constexpr std::bitset<10> isTaskGoalError{         0b00'0000'1000 };
+  static constexpr std::bitset<10> isCamGoalError{          0b00'0001'0000 };
+  static constexpr std::bitset<10> isCamExecutionError{     0b00'0010'0000 };
+  static constexpr std::bitset<10> isPanTiltGoalError{      0b00'0100'0000 };
+  static constexpr std::bitset<10> isPanTiltExecutionError{ 0b00'1000'0000 };
+  static constexpr std::bitset<10> isLanderExecutionError{  0b01'0000'0000 };
+  static constexpr std::bitset<10> isPowerSystemFault{      0b10'0000'0000 };
```

Comment:
@lanseafood these lines seem to repeat in at least two PR. I think it would be a good idea that you have actually made them part of the `OCEANWATER-719-new_detection_node` branch and then created the two PRs #145 and #146

Reply:
Yea i think the way i set up the initial branches was a little disorganized. Given they're the same though, and wouldn't have merge conflicts if merged in order could we leave it for now and just once both are merged to detection node and detection node is merged to the main PR branch, I can start fresh with the future PRs?

Reply:
I would recommend that you do so. Let's first focus on the `OCEANWATER-719-new_detection_node` which I believe still need work to isolate the fault-detection-node from the fault-injection-node, and you can factor out the shared parts of that PR. Then we can proceed with these PRs.